src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -296,6 +348,9 @@ namespace WalkingControllers
          * @return true/false in case of success/failure
          */
         bool UpdateDCMEstimator(const iDynTree::Vector2 &CoM2DPosition, const iDynTree::Vector2 &CoMVelocity, const iDynTree::Vector2 &measuredZMP, const double &CoMHeight);
+
+        bool runStepAdaptation(const runningStepAdapterInput input, runStepAdapterOutput& output, std::unique_ptr<TrajectoryGenerator> trajectoryGeneratorStepAdjustment, std::unique_ptr<TrajectoryGenerator> trajectoryGenerator);
```

Comment:
Hi @S-Dafarra. I tried to move  the step adjustment part from walkingModule to step adaptation library.
However, I have multiple error after running the walking module. Would you please tell me is it right to have a method  with these input and output inside the `stepAdaptationController` Library?

Reply:
Hi @MiladShafiee,  passing a ``std::unique_ptr`` is not a good idea. The idea behind ``unique_ptr`` is that they are uniquely own by the ``WalkingModule`` class. Like your liver belongs to you :grin: If you pass your liver to someone else (with ``std::move``), you won't be able to use it anymore. Similarly, if you ``move`` the ``TrajectoryGenerator`` object to the ``StepAdaptation`` library, the ``WalkingModule`` cannot use it anymore. To be honest, I don't have a clean solution to move everything to the step adaptation library. As I anticipated you during a call, a possibility may be to simply create a bunch of new methods inside the ``WalkingModule``, but at least we remove the step adaptation logic out of the ``update`` loop.

Reply:
Thanks! I was struggeling with this and I was thinking that `move` magically solve this problem that it seems it's wrong!
What do you think about to keep just the trajectory re-planning part outside `stepAdaptationController` library?

Reply:
> What do you think about to keep just the trajectory re-planning part outside stepAdaptationController library?

As you prefer. Again, right now I don't see a very clean solution. You can start creating a simple function within the ``Module``. Later on, we may check if some other solution may be available.